### PR TITLE
Hotfix/speakers activity count tweaks

### PR DESCRIPTION
**Pull request by smarcet**
ref: https://app.clickup.com/t/9014802374/86b9b1qrk?comment=90140217991206

### `getUniqueActivitiesCountBySummit` — Speaker & Submitter repos

**Before:** correlated `EXISTS` wrapping a nested `IN (SELECT DISTINCT ...)` subquery — O(presentations × speakers/members) execution plan.
Measured at **12–14 s** on dev data.

**After:** two-phase temp-table pattern:
- Phase 1 — DQL QB (same filter system) streams matching IDs into a session-scoped `MEMORY` temp table in chunks of 1 000 via `LIMIT/OFFSET` + `INSERT IGNORE`. PHP never holds the full set in memory.
- Phase 2 — raw SQL `JOIN` against the indexed temp table; no correlated subqueries, no large `IN` array passed over the wire.
- `try/finally` guarantees `DROP TEMPORARY TABLE` even on exception.

Result: query completes in **< 1 s** on the same dev dataset.

### `getAllByPage` (speakers) and `getSubmittersBySummit`

**Before:** single large query hydrating full entities including all associations in one pass — slow and memory-intensive on summits with many speakers/submitters.

**After:** two-phase pagination:
- Phase 1 — `COUNT(DISTINCT e.id)` for the total, then `SELECT DISTINCT e.id` with `LIMIT/OFFSET` for the current page.
- Phase 2 — `WHERE e.id IN (:ids)` to hydrate only the current page of entities.

This eliminates the cost of counting and sorting full entity graphs; the DB optimises count and ID queries independently from hydration.

---

## Summary by CodeRabbit

* **New Features**
  * Added support for filtering summit submitters by `presentations_track_group_id` across submitter listing, CSV export, bulk email, and unique-activity counting.
* **Refactor**
  * Improved submitter and speaker listing pagination and ordering, with more consistent paging across pages.
* **Bug Fixes**
  * Fixed and stabilized unique-activity counting for submitters and speakers to correctly honor filters, including moderator-related activity.
* **Tests**
  * Expanded coverage for paging/disjoint pagination, new track-group filter behavior, and unique-activity counting edge cases.

### CODE REVIEW

File: app/Repositories/Summit/DoctrineMemberRepository.php [hunk 1]
```diff
@@ -575,67 +575,107 @@ public function getByEmailExclusiveLock($email): ?Member
...
-                ->setParameter("summit", $summit);
-        },
-            $paging_info,
-            $filter,
-            $order,
-            function ($query) {
-                //default order
-                return $query->addOrderBy("e.id", 'ASC');
-            });
+        return $this->getSubmittersIdsByPage($summit, $paging_info, $filter, $order);
```
[the remark this answers is not in the document]
Comment (smarcet) [reply app/Repositories/Summit/DoctrineMemberRepository.php:699]: False positive — the old implementation via `getParametrizedAllIdsByPage` (DoctrineRepository.php:234) also returned a plain `array` (not a `PagingResponse`). The `@return PagingResponse` on the interface was an incorrect annotation pre-dating this PR. The single consumer (`ParametrizedSendEmails` trait) iterates with `count()` and `foreach` — designed for a plain array. No behaviour change here. Fixing the interface annotation in this PR.

**Review by romanetar — APPROVED** [review]
LGTM
